lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -268,13 +293,19 @@ public boolean isRetryDelayValid()
         return minBackoffDelay.compareTo(maxBackoffDelay) <= 0;
     }
 
-    @AssertTrue(message = "Either gcs.use-access-token or gcs.json-key or gcs.json-key-file-path must be set")
+    @AssertTrue(message = "Either gcs.auth-type or gcs.json-key or gcs.json-key-file-path must be set")
     public boolean isAuthMethodValid()
     {
-        if (useGcsAccessToken) {
+        if (getAuthType() == AuthType.ACCESS_TOKEN) {
             return jsonKey == null && jsonKeyFilePath == null;
         }
 
         return (jsonKey == null) ^ (jsonKeyFilePath == null);
     }
+
+    @AssertTrue(message = "Cannot set both gcs.use-access-token and gcs.auth-type")
+    public boolean isAuthTypeAndUseGcsAccessTokenMutuallyExclusive()
```

Comment:
The name of this method reads funny. It sounds like a question whose answer should always be true. Rename it to `areAuthTypeAndGcsAccessTokenConfigured` and use AssertFalse

Reply:
I think @wendigo is right, the validation method needs to start with "is", otherwise the test in `TestGcsFileSystemConfig` will fail due to validation, see the change https://github.com/trinodb/trino/pull/26681/commits/5512447feb53a6ff97b35be8b53bdc1f94390e6c

Reply:
Ok, then let's rename it to `isAuthTypeAndGcsAccessTokenConfigured`